logic/getMatchLogic.js

```diff
@@ -0,0 +1,97 @@
+'use strict'
+const MongoClient = require('mongodb').MongoClient
+const MsLogger = require('@first-lego-league/ms-logger').Logger()
+
+const MONGU_URI = process.env.MONGO_URI
+
+const { MClient } = require('mhub')
+
+const mhubClient = new MClient(process.env.MHUB_URI)
+
+if (process.env.DEV) {
+  mhubClient.login('default', '')
+} else {
+  mhubClient.login('protected-client', process.env.PROTECTED_MHUB_PASSWORD)
+}
+
+let match = 0
+const UPCOMING_MATCHES_TO_GET = 2
+let canUpdateMatch = true
+
+mhubClient.connect().then(() => {
+  if (process.env.DEV) {
+    mhubClient.subscribe('default')
+  } else {
+    mhubClient.subscribe('protected-client')
+  }
+})
+
+mhubClient.on('message', msg => {
+  switch (msg.topic) {
+    case 'clock:start':
+      if (canUpdateMatch) {
+        match++
+        canUpdateMatch = false
+      }
+      break
+    case 'clock:end':
+      canUpdateMatch = true
+      break
+  }
+})
+
+function getCurrentMatch () {
+  return getMatch(match)
+}
+
+function getNextMatches () {
+  const retMatches = []
+
+  for (let i = 1; i <= UPCOMING_MATCHES_TO_GET; i++) {
+    retMatches.push(getMatch(match + i))
+  }
+
+  if (retMatches.length > 0) {
+    return retMatches
+  }
+  return null
+}
+
+function getMatchNumber () {
+  return match
+}
+
+function setMatchNumber (newMatch) {
+  match = newMatch
+}
+
+const getMatch = function (matchNumber) {
```

Comment:
Are you sure this is the right place for this logic? Getting a match from the db could be usefull in more places

Reply:
Should I make this function also be exported then, or have it go through some router?

Reply:
I think this logic can  be in a new file called matchLogic and returns to you the full match.
Than you can change the object you get back to whatever you need